pkg/ccl/roleccl/role.go

```diff
@@ -86,16 +86,18 @@ func grantRolePlanHook(
 
 		if hasAdminRole, err := p.HasAdminRole(ctx); err != nil {
 			return err
-		} else if !hasAdminRole {
-			// Not a superuser: check permissions on each role.
+		} else {
+			// Check permissions on each role.
 			allRoles, err := p.MemberOfWithAdminOption(ctx, p.User())
 			if err != nil {
 				return err
 			}
 			for _, r := range grant.Roles {
-				if isAdmin, ok := allRoles[string(r)]; !ok || !isAdmin {
-					return pgerror.Newf(pgcode.InsufficientPrivilege,
-						"%s is not a superuser or role admin for role %s", p.User(), r)
+				if !hasAdminRole || r == "admin" {
```

Comment:
what is this `admin` string? can it be a constant? maybe add a comment?

Reply:
The constant for the name of the `admin` role is `sqlbase.AdminRole`.
I agree with Matt, this needs some comments. I would also make it more explicit by inverting the condition and `continue`ing on to the next loop.